clarity/src/vm/analysis/type_checker/tests/contracts.rs

```diff
@@ -630,3 +664,1864 @@ fn test_expects() {
+                }
+            }
+            _ => false,
+        });
+        let err = mem_type_check_v1(if_branches_with_incompatible_trait_types).unwrap_err();
+        eprintln!(
+            "Clarity1: if branches with incompatible trait types: {}",
+            err
+        );
+        assert!(match err {
+            CheckError {
+                err: CheckErrors::IfArmsMustMatch(type1, type2),
+                expressions: _,
+                diagnostic: _,
+            } => {
+                match (type1, type2) {
+                    (
+                        TypeSignature::CallableType(CallableSubtype::Trait(trait1)),
+                        TypeSignature::CallableType(CallableSubtype::Trait(trait2)),
+                    ) => {
+                        assert_eq!(trait1.name.as_str(), "trait-1");
+                        assert_eq!(trait2.name.as_str(), "trait-2");
+                        true
+                    }
+                    _ => false,
+                }
+            }
+            _ => false,
+        });
+
+        let err = mem_type_check(if_branches_with_compatible_trait_types).unwrap_err();
+        eprintln!("if branches with compatible trait types: {}", err);
+        assert!(match err {
+            CheckError {
+                err: CheckErrors::IfArmsMustMatch(type1, type2),
+                expressions: _,
+                diagnostic: _,
+            } => {
+                match (type1, type2) {
+                    (
+                        TypeSignature::CallableType(CallableSubtype::Trait(trait1)),
+                        TypeSignature::CallableType(CallableSubtype::Trait(trait2)),
+                    ) => {
+                        assert_eq!(trait1.name.as_str(), "trait-1");
+                        assert_eq!(trait2.name.as_str(), "trait-2");
+                        true
+                    }
+                    _ => false,
+                }
+            }
+            _ => false,
+        });
+        let err = mem_type_check_v1(if_branches_with_compatible_trait_types).unwrap_err();
+        eprintln!("Clarity1: if branches with compatible trait types: {}", err);
+        assert!(match err {
+            CheckError {
+                err: CheckErrors::IfArmsMustMatch(type1, type2),
+                expressions: _,
+                diagnostic: _,
+            } => {
+                match (type1, type2) {
+                    (
+                        TypeSignature::CallableType(CallableSubtype::Trait(trait1)),
+                        TypeSignature::CallableType(CallableSubtype::Trait(trait2)),
+                    ) => {
+                        assert_eq!(trait1.name.as_str(), "trait-1");
+                        assert_eq!(trait2.name.as_str(), "trait-2");
+                        true
+                    }
+                    _ => false,
+                }
+            }
+            _ => false,
+        });
+    }
+}
+
+/// Based on issue #3215 from sskeirik
```

Comment:
Thanks for folding all of @sskeirik's tests in!

Reply:
In some of these cases, would it make sense to verify that the buggy behavior in Clarity1 is preserved in these test vectors?  I think it would help build confidence that the new trait handler is bug-for-bug compatible with the old one.

Reply:
+1.. would it make sense to do this?